### PR TITLE
feat: add multi tabs

### PR DESCRIPTION
**Pull request by benjaminlim00**

### CODE REVIEW

File: src/components/document-renderer.tsx [hunk 1]
```diff
@@ -40,13 +54,30 @@ export const DocumentRenderer: React.FunctionComponent<DocumentRendererProps> =
     }
   }, [document, toFrame]);
 
+  useEffect(() => {
+    if (toFrame && selectedTemplate) {
+      // TODO: how can i get this to run? selected template is updated on new tab click but toFrame is null so it does not run.
+      toFrame({
```
Comment (benjaminlim00) [inline src/components/document-renderer.tsx:60]: When I click on one of the tabs, selectedTemplate will be changed and i should run toFrame to update the template. But how can i get toFrame to not have a value of null? i don't really understand how toFrame is used

File: src/components/document-renderer.tsx [hunk 2]
```diff
@@ -10,19 +11,32 @@ export const DocumentRenderer: React.FunctionComponent<DocumentRendererProps> =
   rawDocument,
 }: DocumentRendererProps) => {
   const document = getData(rawDocument);
+  const [templates, setTemplates] = useState([]);
```
Comment (Nebulis) [inline src/components/document-renderer.tsx:14]: create a type for templates

File: src/components/document-renderer.tsx [hunk 3]
```diff
@@ -10,19 +11,32 @@ export const DocumentRenderer: React.FunctionComponent<DocumentRendererProps> =
...
+  const [selectedTemplate, setSelectedTemplate] = useState("");
+
+  const updateTemplates = useCallback((templates) => {
+    setTemplates(templates);
+    setSelectedTemplate(templates[0].id);
+  }, []);
 
   const [toFrame, setToFrame] = useState<HostActionsHandler | null>();
-  const [height, setHeight] = useState();
+  const [height, setHeight] = useState<number>();
```
Comment (Nebulis) [inline src/components/document-renderer.tsx:23]: initial value => 0

File: src/components/document-renderer.tsx [hunk 4]
```diff
@@ -40,13 +54,30 @@ export const DocumentRenderer: React.FunctionComponent<DocumentRendererProps> =
...
   return (
-    <FrameConnector
-      style={{ height: `${height}px`, width: "100%", border: "0px" }}
-      source={document.$template.url}
-      onConnected={onConnected}
-      dispatch={fromFrame}
-      key={document.id} // we will reset the connection every time the id change (react will mount a new component)
-    />
+    <>
+      <MultiTabs
```
Comment (Nebulis) [inline src/components/document-renderer.tsx:69]: to hide if no templates (templates.length > 0)

File: src/components/multi-tabs.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
+import React from "react";
+
+interface MultiTabsProps {
+  templates: { id: string; label: string }[];
+  selectedTemplate: string;
+  onSelectTemplate: (id: string) => void;
+}
+
+export const MultiTabs: React.FunctionComponent<MultiTabsProps> = ({
```
Comment (Nebulis) [inline src/components/multi-tabs.tsx:9]: Just Tabs ? it already means Multi by itself :thinking:

File: src/pages/home.tsx [hunk 6]
```diff
@@ -41,7 +41,10 @@ export const HomePage: React.FunctionComponent = () => {
           <p className="text-red-600 font-medium">Verifiable Credential is invalid</p>
         </div>
       ) : null}
-      {documentStatus === Status.RESOLVED ? <DocumentRenderer rawDocument={rawDocument as WrappedDocument} /> : null}
+      {/* {documentStatus === Status.RESOLVED ? <DocumentRenderer rawDocument={rawDocument as WrappedDocument} /> : null} */}
+      {documentStatus !== Status.PENDING && documentStatus !== Status.IDLE ? (
```
Comment (Nebulis) [inline src/pages/home.tsx:45]: why changing this ?

Comment (benjaminlim00) [reply src/pages/home.tsx]: only for testing to show tabs using opencerts file with tabs available

Comment (benjaminlim00) [reply src/pages/home.tsx]: i have reverted the change

**Pull request merged**